kura/org.eclipse.kura.rest.security.provider/src/main/java/org/eclipse/kura/internal/rest/security/provider/SecurityRestServiceV2.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ ******************************************************************************/
+package org.eclipse.kura.internal.rest.security.provider;
+
+import org.eclipse.kura.request.handler.jaxrs.DefaultExceptionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+@Path("security/v2")
+public class SecurityRestServiceV2 extends AbstractRestSecurityService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SecurityRestServiceV2.class);
+    private static final String MQTT_APP_ID = "SEC-V2";
+
+    @Override
+    public String getMqttAppId() {
+        return MQTT_APP_ID;
+    }
+
+    /**
+     * POST method <br /> This method replaces the security policy with the default production one. Then a fingerprint
+     * reload is performed.
+     */
+    @POST
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/security-policy/apply-default-production")
+    public Response applyDefaultProductionSecurityPolicy() {
+        try {
+            logger.debug(DEBUG_MESSAGE, "applyDefaultProductionSecurityPolicy");
+
+            this.security.applyDefaultProductionSecurityPolicy();
+            this.security.reloadSecurityPolicyFingerprint();
+            this.security.reloadCommandLineFingerprint();
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+        return Response.ok().build();
+    }
+
+    /**
+     * POST method <br /> This method replaces the security policy with the provided one. Then a fingerprint reload is
+     * performed.
+     */
+    @POST
+    @RolesAllowed(REST_ROLE_NAME)
+    @Path("/security-policy/apply")
+    public Response applySecurityPolicy(InputStream securityPolicyInputStream) {
+        try {
+            logger.debug(DEBUG_MESSAGE, "applySecurityPolicy");
+
+            this.security.applySecurityPolicy(readSecurityPolicyString(securityPolicyInputStream));
+            this.security.reloadSecurityPolicyFingerprint();
+            this.security.reloadCommandLineFingerprint();
+        } catch (Exception e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+
+        return Response.ok().build();
+    }
+
+    private String readSecurityPolicyString(InputStream securityPolicyInputStream) throws IOException {
+        if (securityPolicyInputStream == null) {
+            throw new IllegalArgumentException("Security Policy cannot be null or empty");
```

Comment:
Throwing an IllegalArgumentException would result in status code 500, maybe a 400 would be more appropriate here, you should be able to do that using this method https://github.com/eclipse/kura/blob/71def61f3775869ba6fa762deb81789a44c0b73f/kura/org.eclipse.kura.request.handler.jaxrs/src/main/java/org/eclipse/kura/request/handler/jaxrs/DefaultExceptionHandler.java#L77. The same applies for the other exceptions below